iceoryx_utils/include/iceoryx_utils/cxx/helplets.hpp

```diff
@@ -203,6 +205,14 @@ static constexpr uint64_t strlen2(char const (&/*notInterested*/)[SizeValue])
 #define DISCARD_RESULT(expr) auto DISCARD_RESULT_VARIABLE(unusedOnLine, __LINE__) [[gnu::unused]] = expr
 // clang-format on
 
+/// @brief Prints a warning to std::cerr if called on a 32-bit system
+constexpr void printWarningOn32BitSystems()
```

Comment:
This does not belong into `iceoryx_utils` since the utils are a generic library and in this case we do not have any restrictions. The restrictions are coming from posh!

I would suggest the following approach.
1. Write a function in helplets like:
```cpp
constexpr bool isCompiledOn32BitSystem() {
    return INTPTR_MAX == INT32_MAX;
}
```

2. Use the function in helplets to write a custom message like:
```cpp
if (cxx::isCompiledOn32BitSystem) {
   std::cerr << "not supported\n";
}
```
With this approach you can provide the user also a more detailed message like: roudi does not work in 32-bit mode or something like this.